src/main/java/com/epam/esm/giftcertificates/service/GiftCertificatesServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+package com.epam.esm.giftcertificates.service;
+
+import com.epam.esm.giftcertificates.filter.Chain;
+import com.epam.esm.giftcertificates.utils.GiftCertificatesComplement;
+import com.epam.esm.giftcertificates.entity.GiftCertificate;
+import com.epam.esm.giftcertificates.repo.GiftCertificatesRepository;
+import com.epam.esm.giftcertificates.exception.CertificateNotRepresent;
+import com.epam.esm.giftcertificates.exception.CertificateInvalidRequest;
+import com.epam.esm.giftcertificates.exception.CertificateNotFound;
+import com.epam.esm.giftcertificates.validation.GiftValidate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Supplier;
+
+@Service
+public class GiftCertificatesServiceImpl implements GiftCertificatesService {
+
+    private final GiftCertificatesRepository giftCertificatesRepository;
+
+    @Autowired
+    public GiftCertificatesServiceImpl(GiftCertificatesRepository giftCertificatesRepository) {
+        this.giftCertificatesRepository = giftCertificatesRepository;
+    }
+
+    @Override
+    public List<GiftCertificate> readGiftCertificate(Map<String, String> requestmap) {
+        GiftValidate.findGiftByFieldsValidation(
+                requestmap.get("sort_date"), requestmap.get("sort_name"),
+                requestmap.get("tag_name"), requestmap.get("gift_name"),
+                requestmap.get("description"));
+        Chain chain = new Chain(requestmap);
+        return giftCertificatesRepository.getGiftCertificateByParam(chain.buildQuery(), chain.buildParamList())
```

Comment:
`chain.buildQuery(), chain.buildParamList()` Sounds like it is not a chain, but some kind of builder